pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -205,9 +208,20 @@ func (r *ReplicaSetReconciler) isStatefulSetReady(mdb mdbv1.MongoDB, expectedUpd
 	return statefulset.IsReady(set, mdb.Spec.Members) && expectedUpdateStrategy == set.Spec.UpdateStrategy.Type, nil
 }
 
+func (r *ReplicaSetReconciler) createService(mdb mdbv1.MongoDB) error {
+	svc := buildService(mdb)
+	svcCopy := svc.DeepCopyObject()
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.Name, Namespace: mdb.Namespace}, svcCopy)
```

Comment:
watch out! the name of the service is not the same as the name of the MongoDB resource, it is `<mdb-name>-svc`, there is a function `mdb.ServiceName()` we can use here.

Reply:
Ouch! Thanks for spotting this!